apps/web/src/pages/SearchRequests/SearchPage/utils.ts

```diff
@@ -0,0 +1,234 @@
+import { IntlShape, MessageDescriptor } from "react-intl";
+
+import {
+  unpackMaybes,
+  empty,
+  hasKey,
+  notEmpty,
+  pickMap,
+} from "@gc-digital-talent/helpers";
+import {
+  ApplicantFilterInput,
+  Classification,
+  CountApplicantsQueryVariables,
+  Maybe,
+  Pool,
+  PositionDuration,
+} from "@gc-digital-talent/graphql";
+import { EmploymentDuration } from "@gc-digital-talent/i18n";
+
+import { SimpleClassification } from "~/types/pool";
+import { FormValues, NullSelection } from "~/types/searchRequest";
+import {
+  formatClassificationString,
+  poolMatchesClassification,
+} from "~/utils/poolUtils";
+import nonExecutiveITClassifications from "~/constants/nonExecutiveITClassifications";
+import { positionDurationToEmploymentDuration } from "~/utils/searchRequestUtils";
+
+export const getAvailableClassifications = (
+  pools: Pool[],
+): Classification[] => {
+  const availableClassifications = pools
+    ?.flatMap((pool) => pool?.classifications)
+    .filter(notEmpty)
+    .reduce((currentClassifications, classification) => {
+      let newClassifications = [...currentClassifications];
+      const includesClassification = newClassifications.find(
+        (c) => c.id === classification.id,
+      );
+      if (!includesClassification) {
+        newClassifications = [...newClassifications, classification];
+      }
+
+      return newClassifications;
+    }, [] as Classification[]);
+
+  const ITClassifications = nonExecutiveITClassifications();
+
+  return availableClassifications.filter((classification) => {
+    return ITClassifications.some(
+      (x) =>
+        x?.group === classification?.group &&
+        x?.level === classification?.level,
+    );
+  });
+};
+
+export const getClassificationLabel = (
+  { group, level }: Classification,
+  labels: Record<string, MessageDescriptor>,
+  intl: IntlShape,
+) => {
+  const key = `${group}-0${level}`;
+  return !hasKey(labels, key) ? key : intl.formatMessage(labels[key]);
+};
+
+/**
+ * Derive the currently selected classification
+ * from applicant filters and location state.
+ *
+ * As well as transforming it to a useable string.
+ *
+ * @param {ApplicantFilterInput} data
+ * @param {Maybe<SimpleClassification[]>} selectedClassifications
+ * @returns {string}
+ */
+const getCurrentClassification = (
+  data: ApplicantFilterInput,
+  selectedClassifications?: Maybe<SimpleClassification[]>,
+): string => {
+  const classifications = data?.qualifiedClassifications?.filter(notEmpty);
+  const classification = selectedClassifications ?? classifications;
```

Comment:
Is this a existing functionality? 
CurrentClassification is existing government employees job classification.
We are assigning it here with qualified pool candidates classification 🤔

Reply:
Oh yeah, looks like this is not supposed to use `data` at all. 

https://github.com/GCTC-NTGC/gc-digital-talent/blob/3bc3a032cfc6f6f8887f91e8d1829224e5b2a053/apps/web/src/pages/SearchRequests/SearchPage/components/SearchForm.tsx#L66C33-L68

Reply:
Updated in [do not derive classification from filter data](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8340/commits/5cb1960156e68c24aceefd56fbcd14a5c1dcc635)